### PR TITLE
implement camera bobing and breathing

### PR DESCRIPTION
**Pull request by elsonel**
still working on refining movement script, but did another little polish in the meantime.

- implemented view bobing
- implemented view breathing

https://user-images.githubusercontent.com/10892740/195332924-fe23036e-db51-410d-b1fa-9615d0ac8dbe.mp4

### CODE REVIEW

File: Assets/Scripts/CameraBob.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private float camInitialLocalY;
+    private int breathFrameCounter = 0;
+    private int bobFrameCounter = 0;
+    public bool isBobing = false;
+    private bool isBreathing = true;
+    public float currentOffset = 0;
+
+
+    void OnValidate() {
+        breathIntensity = Mathf.Max(0, breathIntensity);
```
Comment (c-boyle) [inline Assets/Scripts/CameraBob.cs:26]: You can use the `PositiveValueOnly` attribute from MyBox to do this too (I haven't installed the package yet though): https://github.com/Deadcows/MyBox/wiki/Attributes#maxvalue-minvalue-and-positivevalueonly

**Review by c-boyle — APPROVED** [review]
Besides a tiny nitpick suggestion for the validation, it all looks good to me code wise. I do think the bob settings may need a little tweaking from the video though just to make it look a bit smoother

File: Assets/Scripts/CameraBob.cs (hunk 1 above)
Comment (elsonel) [reply Assets/Scripts/CameraBob.cs]: Resolved

**Comment by elsonel** [comment]
@c-boyle I think the view bobbing looks fine in terms of camera movement only. As a whole it looks weird because of the arm's position in reference. 

When I get time later, I'll fix the relationship between the arm and camera and it should look better.

**Pull request merged**